core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -597,6 +597,24 @@ public TableMetadata replaceProperties(Map<String, String> rawProperties) {
         .build();
   }
 
+  /**
+   * Prune the unused partition specs from the table metadata.
+   *
+   * <p>Note: it's not safe for external client to call this directly, it's usually called by the
```

Comment:
This method is package private, does it need this note?

Reply:
I agree with @RussellSpitzer. This class is internal and many of its methods can break tables if called incorrectly.

Also, we are no longer adding new operation methods to this. These days we make modifications to the `Builder` instead.